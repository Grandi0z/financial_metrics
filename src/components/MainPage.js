import React from 'react';
import { Link } from 'react-router-dom';
import FilterCompanyProfiles from './FilterCompanyProfiles';

const MainPage = () => (
  <>
    <div className="region">
      <div className="bar">
        <div>
          <Link to="/"><i className="bi bi-chevron-left" /></Link>
          <span>{new Date().getFullYear()}</span>
        </div>
        <span>most Companies</span>
        <div className="navLinks">
          <Link href="/#"><i className="bi bi-mic-fill" /></Link>
          <Link href="/#"><i className="bi bi-gear-wide" /></Link>
        </div>
      </div>
      <div className="bar-desk">
        <div>
          <Link to="/" style={{ textDecoration: 'none' }}>Companies</Link>
        </div>
        <span>most Companies</span>
        <div className="navLinks">
          <Link href="/#" style={{ textDecoration: 'none' }}>
            <i className="bi bi-gear-wide" />
            Settings
          </Link>
        </div>
      </div>
      <h1 className="country">42 USA&apos;S COMPANIES</h1>
    </div>
    <FilterCompanyProfiles />
  </>
);

export default MainPage;
