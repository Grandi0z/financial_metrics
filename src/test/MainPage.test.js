import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect'
import MainPage from '../components/MainPage'
import companiesReducer from '../redux/features/mainPage/companiesSlice';
import companyDetailsReducer from '../redux/features/companyDetails/companyDetailsSlice';
import { MemoryRouter } from 'react-router-dom'

const store = configureStore({
    reducer: {
        companies: companiesReducer,
        companyDetails: companyDetailsReducer,
    },
  });
  
  describe('Main Page', () => {
    it('Should match the snapshot', () => {
      const tree = renderer.create(
        <Provider store={store}>
            <MemoryRouter><MainPage /></MemoryRouter>
        </Provider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });