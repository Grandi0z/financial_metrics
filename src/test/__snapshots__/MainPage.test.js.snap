// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page Should match the snapshot 1`] = `
Array [
  <div
    className="region"
  >
    <div
      className="bar"
    >
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="bi bi-chevron-left"
          />
        </a>
        <span>
          2023
        </span>
      </div>
      <span>
        most Companies
      </span>
      <div
        className="navLinks"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="bi bi-mic-fill"
          />
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="bi bi-gear-wide"
          />
        </a>
      </div>
    </div>
    <h1
      className="country"
    >
      42 USA'S COMPANIES
    </h1>
  </div>,
  <div>
    Data are fetching...
  </div>,
]
`;
